hnix-store-core/src/System/Nix/ReadonlyStore.hs

```diff
@@ -6,27 +6,37 @@ module System.Nix.ReadonlyStore where
 
 import           Data.ByteString (ByteString)
 import qualified Data.ByteString as BS
+import qualified Data.Text as T
 import qualified Data.HashSet as HS
 import           Data.Text.Encoding
 import           System.Nix.Hash
 import           System.Nix.StorePath
 
-makeStorePath :: forall storeDir hashAlgo . (KnownStoreDir storeDir, NamedAlgo hashAlgo) => ByteString -> Digest hashAlgo -> StorePathName -> StorePath storeDir
-makeStorePath ty h nm = StorePath storeHash nm
+makeStorePath :: forall hashAlgo . (NamedAlgo hashAlgo) => FilePath -> ByteString -> Digest hashAlgo -> StorePathName -> StorePath
+makeStorePath fp ty h nm = StorePath storeHash nm fp
   where
     s = BS.intercalate ":"
       [ ty
       , encodeUtf8 $ algoName @hashAlgo
       , encodeUtf8 $ encodeBase16 h
-      , storeDirVal @storeDir
+      , encodeUtf8 $ T.pack fp
       , encodeUtf8 $ unStorePathName nm
       ]
     storeHash = hash s
 
-makeTextPath :: (KnownStoreDir storeDir) => StorePathName -> Digest 'SHA256 -> StorePathSet storeDir -> StorePath storeDir
-makeTextPath nm h refs = makeStorePath ty h nm
+makeTextPath :: FilePath -> StorePathName -> Digest 'SHA256 -> StorePathSet -> StorePath
+makeTextPath fp nm h refs = makeStorePath fp ty h nm
   where
     ty = BS.intercalate ":" ("text" : map storePathToRawFilePath (HS.toList refs))
 
-computeStorePathForText :: (KnownStoreDir storeDir) => StorePathName -> ByteString -> StorePathSet storeDir -> StorePath storeDir
-computeStorePathForText nm s refs = makeTextPath nm (hash s) refs
+makeFixedOutputPath :: (ValidAlgo hashAlgo, NamedAlgo hashAlgo) => FilePath -> Bool -> Digest hashAlgo -> StorePathName -> StorePath storeDir
```

Comment:
This looks like a stray `StorePath storeDir` to me. Does it compile for you ? I get


```
src/System/Nix/ReadonlyStore.hs:32:124: error:
    • Expected kind ‘k0 -> *’, but ‘StorePath’ has kind ‘*’
    • In the type signature:
        makeFixedOutputPath :: (ValidAlgo hashAlgo, NamedAlgo hashAlgo) =>
                               FilePath
                               -> Bool -> Digest hashAlgo -> StorePathName -> StorePath storeDir
   |
32 | makeFixedOutputPath :: (ValidAlgo hashAlgo, NamedAlgo hashAlgo) => FilePath -> Bool -> Digest hashAlgo -> StorePathName -> StorePath storeDir
   |                                                                                                                            ^^^^^^^^^^^^^^^^^^
cabal: Failed to build hnix-store-core-0.2.0.0 (which is required by
hnix-store-remote-0.2.0.0).
```

Reply:
Rebasing error, should be good now.